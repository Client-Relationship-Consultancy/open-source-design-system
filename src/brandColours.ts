export const colourPalette = {
  examplePalette: {
    primary: {
      main: {
        hex: "#A42824",
        RGB: "164,40,36",
        on: "#FFFFFF",
      },
      light: {
        hex: "#DA342F",
        RGB: "218,52,47",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#661614",
        RGB: "102,22,20",
        on: "#FFFFFF",
      },
    },
    secondary: {
      main: {
        hex: "#0B466B",
        RGB: "11,70,107",
        on: "#FFFFFF",
      },
      light: {
        hex: "#599CAF",
        RGB: "89,156,175",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#05263B",
        RGB: "5,38,59",
        on: "#FFFFFF",
      },
    },
    complimentary: {
      main: {
        hex: "#FFEE52",
        RGB: "255,238,82",
        on: "#272727",
      },
      light: {
        hex: "#FFF59D",
        RGB: "255,245,157",
        on: "#272727",
      },
      dark: {
        hex: "#CEB900",
        RGB: "206,185,0",
        on: "#272727",
      },
    },
    action: {
      main: {
        hex: "#743C8F",
        RGB: "116,60,143",
        on: "#FFFFFF",
      },
      light: {
        hex: "#AD65CF",
        RGB: "173,101,207",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#4E2662",
        RGB: "79,38,98",
        on: "#FFFFFF",
      },
    },
    error: {
      main: {
        hex: "#FF7320",
        RGB: "255,115,32",
        on: "#FFFFFF",
      },
      dark: {
        hex: "#C54901",
        RGB: "197,73,1",
        on: "#FFFFFF",
      },
    },
    background: {
      hex: "#F6F6F6",
      RGB: "246,246,246",
    },
    surface: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    white: {
      hex: "#FFFFFF",
      RGB: "255,255,255",
      on: "#272727",
    },
    black: {
      main: { hex: "#272727", RGB: "39,39,39", on: "#FFFFFF" },
      tint80: { hex: "#696969", RGB: "105,105,105", on: "#FFFFFF" },
      tint60: { hex: "#A0A0A0", RGB: "160,160,160", on: "#FFFFFF" },
      tint40: { hex: "#C7C7C7", RGB: "199,199,199", on: "#272727" },
      tint20: { hex: "#ECECEC", RGB: "236,236,236", on: "#272727" },
    },
  },
};

interface IColourCodes {
  hex: string;
  RGB: string;
  on?: string;
}
interface IColourVariations {
  main: IColourCodes;
  light?: IColourCodes;
  dark?: IColourCodes;
}

interface IBlackVariations {
  main: IColourCodes;
  tint80: IColourCodes;
  tint60: IColourCodes;
  tint40: IColourCodes;
  tint20: IColourCodes;
}

export interface IColourPalette {
  primary: IColourVariations;
  secondary: IColourVariations;
  complimentary: IColourVariations;
  action: IColourVariations;
  error: IColourVariations;
  background: IColourCodes;
  surface: IColourCodes;
  white: IColourCodes;
  black: IBlackVariations;
}
