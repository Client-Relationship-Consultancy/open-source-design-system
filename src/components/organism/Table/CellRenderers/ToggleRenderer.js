import React from "react"
import PropTypes from "prop-types"

import TextToggle from "atoms/TextToggle"

export default class ToggleRenderer extends React.Component {
  handleOnChange = () => {
    this.props.onChange(!this.props.value) // redux change
    this.props.setValue(!this.props.value) // visual change
  }

  render() {
    return (
      <TextToggle
        isDisabled={this.props.isDisabled}
        checked={this.props.value}
        onChange={this.handleOnChange}
        theme={this.props.theme}
      />
    )
  }
}

ToggleRenderer.displayName = "ToggleRenderer"

ToggleRenderer.propTypes = {
  value: PropTypes.bool.isRequired,
  setValue: PropTypes.func.isRequired,
  onChange: PropTypes.func,
  isDisabled: PropTypes.bool,
  theme: PropTypes.object,
}
