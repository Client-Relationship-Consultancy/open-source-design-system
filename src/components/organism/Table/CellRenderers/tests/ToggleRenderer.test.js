import React from "react"
import { shallow } from "enzyme"
import "jest-styled-components"
import { colourPalette } from "../../../../../brandColours"
import ToggleRenderer from "../ToggleRenderer"

const theme = {
  props: {
    theme: colourPalette.examplePalette,
  },
}

describe("ToggleRenderer test", () => {
  it("handleOnChange test", () => {
    const currentValue = false

    const setValue = jest.fn()
    const onChange = jest.fn()

    const component = shallow(
      <ToggleRenderer
        agGridReact={theme}
        setValue={setValue}
        onChange={onChange}
        value={currentValue}
      />,
    )
    component.instance().handleOnChange()

    expect(setValue).toBeCalledWith(!currentValue)
    expect(onChange).toBeCalledWith(!currentValue)
  })
})

describe("Snapshot", () => {
  it("match last snapshot", () => {
    const component = shallow(<ToggleRenderer agGridReact={theme} setValue={() => {}} value />)
    expect(component).toMatchSnapshot()
  })
})
