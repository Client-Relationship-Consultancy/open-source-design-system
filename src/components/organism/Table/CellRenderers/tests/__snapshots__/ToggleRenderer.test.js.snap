// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match last snapshot 1`] = `
<TextToggle
  checked={true}
  falseOption="False"
  isDisabled={false}
  onChange={[Function]}
  trueOption="True"
  width="4.5rem"
/>
`;
